pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	nsList, err := mapper.GetNamespacesForDynakube(ctx, g.apiReader, dk.Name)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	coreLabels := k8slabels.NewCoreLabels(dk.Name, k8slabels.WebhookComponentLabel)
+
+	secret, err := k8ssecret.BuildForNamespace(consts.BootsTrapperInitSecretName, "", data, k8ssecret.SetLabels(coreLabels.BuildLabels()))
+	if err != nil {
+		return err
+	}
+
+	err = k8ssecret.Query(g.client, g.apiReader, log).CreateOrUpdateForNamespaces(ctx, secret, nsList)
+	if err != nil {
+		return err
+	}
+
+	log.Info("done updating init secrets")
+
+	return nil
+}
+
+func (g *BootstrapperInitGenerator) Cleanup(ctx context.Context, namespaces []corev1.Namespace) error {
+	nsList := make([]string, 0, len(namespaces))
+	for _, ns := range namespaces {
+		nsList = append(nsList, ns.Name)
+	}
+
+	return k8ssecret.Query(g.client, g.apiReader, log).DeleteForNamespaces(ctx, consts.BootsTrapperInitSecretName, nsList)
+}
+
+// generate gets the necessary info the create the init secret data
+func (g *BootstrapperInitGenerator) generate(ctx context.Context, dk *dynakube.DynaKube) (map[string][]byte, error) {
+	hostMonitoringNodes, err := g.getHostMonitoringNodes(dk)
+	if err != nil {
+		return nil, err
+	}
+
+	secretConfig, err := g.createSecretConfigForDynaKube(ctx, dk, hostMonitoringNodes)
+	if err != nil {
+		return nil, err
+	}
+
+	agCerts, err := dk.ActiveGateTLSCert(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	trustedCAs, err := dk.TrustedCAs(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	pmcSecret, err := g.preparePMCSecretForBootstrapper(ctx, *dk)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return map[string][]byte{
+		pmc.InputFileName:        pmcSecret,
+		ca.TrustedCertsInputFile: trustedCAs,
+		ca.AgCertsInputFile:      agCerts,
+		curl.InputFileName:       []byte(strconv.Itoa(secretConfig.InitialConnectRetry)),
+		endpoint.InputFileName:   []byte(secretConfig.ApiToken),
+	}, nil
+}
+
+func (g *BootstrapperInitGenerator) preparePMCSecretForBootstrapper(ctx context.Context, dk dynakube.DynaKube) ([]byte, error) {
+	pmc, err := g.dtClient.GetProcessModuleConfig(ctx, 0)
+	if err != nil {
+		conditions.SetDynatraceApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+		return nil, err
+	}
+
+	tenantToken, err := k8ssecret.GetDataFromSecretName(ctx, g.apiReader, types.NamespacedName{
+		Name:      dk.OneAgent().GetTenantSecret(),
+		Namespace: dk.Namespace,
+	}, connectioninfo.TenantTokenKey, log)
+	if err != nil {
+		conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+		return nil, err
+	}
+
+	pmc = pmc.
+		AddHostGroup(dk.OneAgent().GetHostGroup()).
+		AddConnectionInfo(dk.Status.OneAgent.ConnectionInfoStatus, tenantToken).
+		// set proxy explicitly empty, so old proxy settings get deleted where necessary
+		AddProxy("")
```

Comment:
possible **future** improvement

we should probably move such functions to the [bootstrapper-repo](https://github.com/Dynatrace/dynatrace-bootstrapper/blob/main/pkg/configure/oneagent/pmc/ruxit/types.go) as we have to work with these there.